src/app/dataset-view/additional-components/overview-component/overview-component.ts

```diff
@@ -84,4 +117,48 @@ export class OverviewComponent extends BaseComponent implements OnInit {
             ? this.currentState.overview.metadata.chain.blocks.nodes[0]
             : undefined;
     }
+
+    public openInformationModal(content: unknown) {
+        this.modalService.open(content, { centered: true });
+    }
+
+    public commitSetInfoEvent(): void {
+        if (this.datasetBasics) {
+            this.trackSubscription(
+                this.createDatasetService
+                    .commitEventToDataset(
+                        this.datasetBasics.owner.name,
+                        this.datasetBasics.name as string,
+                        this.buildYamlSetInfoEvent(),
+                    )
+                    .subscribe(() => {
+                        this.yamlSetInfoTemplate = this.initialTemplate;
+                    }),
+            );
+        }
+    }
+
+    private buildYamlSetInfoEvent(): string {
+        if (this.description)
+            this.yamlSetInfoTemplate += `  description: ${this.description}\n`;
+        if (this.keywords.length) {
+            this.yamlSetInfoTemplate += `  keywords:\n`;
+            this.keywords.forEach(
+                (keyword: string) =>
+                    (this.yamlSetInfoTemplate += `    - ${keyword}\n`),
+            );
+        }
+        return this.yamlSetInfoTemplate;
+    }
```

Comment:
This does not belong to visual components. Maybe a separate utility service.

Reply:
Moved this method to separate service